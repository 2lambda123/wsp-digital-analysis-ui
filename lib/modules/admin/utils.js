// @flow
import {connect} from 'react-redux'
import {createStructuredSelector} from 'reselect'

import * as actions from './actions'
import * as select from './selectors'

export function fullyConnect (Component: any) {
  return connect(
    createStructuredSelector(select),
    actions
  )(Component)
}
