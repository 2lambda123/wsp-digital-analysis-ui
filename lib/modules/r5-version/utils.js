//
import {VERSION_PARSE_REGEX} from './constants'

/**
 * Convert an r5 version to a number for comparing
 */
export function versionToNumber(version) {
  const versionMatch = (version || '').match(VERSION_PARSE_REGEX)
  if (versionMatch) {
    const [, major, minor, patch, commit] = versionMatch

    // NB Assumes no major, minor, patch, or commit version exceeds 100
    return (
      parseInt(major) * 10000 +
      parseInt(minor) * 100 +
      parseInt(patch) +
      parseInt(commit || 0) / 100
    )
  }

  return 0
}
