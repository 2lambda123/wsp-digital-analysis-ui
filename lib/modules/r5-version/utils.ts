import {MINIMUM_R5_VERSION, VERSION_PARSE_REGEX} from './constants'

/**
 * Convert an r5 version to a number for comparing
 */
export function versionToNumber(version: string): number {
  const versionMatch = (version || '').match(VERSION_PARSE_REGEX)
  if (versionMatch) {
    const [, major, minor, patch, commit] = versionMatch

    // NB Assumes no major, minor, patch, or commit version exceeds 100
    return (
      parseInt(major) * 10000 +
      parseInt(minor) * 100 +
      parseInt(patch) +
      parseInt(commit || '0') / 100
    )
  }

  return 0
}

/**
 * Check if the r5 version is older than a given minimum.
 */
export function workerVersionInRange(
  version: string,
  min: string = MINIMUM_R5_VERSION,
  max = 'v100'
): boolean {
  const versionNumber = versionToNumber(version)
  return (
    versionNumber >= versionToNumber(min) &&
    versionNumber <= versionToNumber(max)
  )
}
