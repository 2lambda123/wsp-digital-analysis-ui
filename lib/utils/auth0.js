// @flow
import message from '@conveyal/woonerf/message'
import get from 'lodash/get'

const CLIENT_ID = process.env.AUTH0_CLIENT_ID
const DOMAIN = process.env.AUTH0_DOMAIN

let LogRocket = {identify () {}}
if (process.env.LOGROCKET) {
  LogRocket = require('logrocket')
  LogRocket.init(process.env.LOGROCKET)
}

// If client & domain exist, then require authentication
export const authIsRequired: boolean = !!(CLIENT_ID && DOMAIN)

// Default Conveyal logo & color
const logo = 'https://d2f1n6ed3ipuic.cloudfront.net/conveyal-128x128.png'
const primaryColor = '#2389c9'

// The default Auth0 scope to request
const scope = 'openid email profile analyst offline_access'

// Default Auth0 response type
const responseType = 'code id_token token'

/**
 * Initialize in a function call so that the Auth0 lock code isn't executed
 * until needed. When Auth0 code is "required" it saves the origin. This is a
 * problem when a user goes to another part of the site (ex: `/regions/create`)
 * and we want to `checkSession` before allowing them to continue. Each "origin"
 * must be saved in the Auth0 Dashboard (which is impossible due to IDs). So
 * instead we redirect to `/login`, `checkSession`, and redirect back to the
 * original path given.
 */
function initializeLock () {
  const Auth0Lock = require('auth0-lock').default
  return new Auth0Lock(CLIENT_ID, DOMAIN, {
    auth: {
      params: {
        responseType,
        scope
      },
      redirect: false
    },
    allowSignUp: false,
    theme: {
      logo,
      primaryColor
    },
    closable: false,
    languageDictionary: {
      title: message('authentication.logIn')
    }
  })
}

/**
 * Initialize Auth0, check the user session, and prompt for a login if needed
 */
export function login (setUser: (any) => void) {
  const lock = initializeLock()

  function handleAuthResult (error, authResult) {
    if (error) {
      return lock.show({
        flashMessage: {
          type: 'error',
          text: error.error_description
        }
      })
    }
    if (!authResult) return lock.show()

    // Flatten the user object
    const user = {
      ...authResult,
      ...authResult.idTokenPayload // contains email, name, nickname, stored metadata
    }

    // Check for the access group and show error if it does not exist
    const accessGroup = get(user, 'analyst.group')
    if (typeof accessGroup !== 'string') {
      return lock.show({
        flashMessage: {
          type: 'error',
          text: `User ${user.email} does not have access to this application.`
        }
      })
    }

    // Log user activities
    LogRocket.identify(user.userId || user.user_id, {
      accessGroup,
      email: user.email
    })

    // AuthResult OK, pass result on. idTokenPayload
    setUser(user)
    lock.hide()
  }

  // Set the default handler for auth errors
  lock.on('authorization_error', handleAuthResult)
  lock.on('authenticated', (authResult) => handleAuthResult(null, authResult))
  lock.on('unrecoverable_error', handleAuthResult)

  // Check the session by default
  lock.checkSession({
    responseType,
    scope,
    timeout: '2000'
  }, handleAuthResult)
}

/**
 * Log out and reload the page.
 */
export function logout () {
  const lock = initializeLock()

  lock.logout({
    clientID: CLIENT_ID,
    returnTo: window.location.origin
  })
}
