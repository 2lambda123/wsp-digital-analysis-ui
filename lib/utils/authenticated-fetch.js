import dbg from 'debug'
import fetch from 'isomorphic-fetch'
import store from '../store' // TODO: don't import store from anywhere but the root/bootstrap

const debug = dbg('scenario-editor:authenticated-fetch')

export default async function authenticatedFetch (url, options = {}) {
  debug(`${options.method || 'get'} ${url.split('?')[0]}`)
  const headers = options.headers || {}
  const state = store.getState()
  if (state.user && state.user.id_token) {
    headers.Authorization = `bearer ${state.user.id_token}`
  }
  const response = await fetch(url, Object.assign({}, options, {headers}))
  debug(`${options.method || 'get'} ${url.split('?')[0]} is ${response.status} ${response.statusText}`)
  return response
}

export function parseJSON (res) {
  if (res.ok) {
    return res.json()
  } else {
    window.alert(res.statusText)
    throw new Error(res.statusText)
  }
}
