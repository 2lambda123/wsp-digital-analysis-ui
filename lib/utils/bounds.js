// @flow
import message from '@conveyal/woonerf/message'
import {LatLng, LatLngBounds} from 'leaflet'

import reprojectCoordinates from './reproject-coordinates'

import type {Bounds} from '../types'

const ANTIMERIDIAN_WARNING = 'We do not currently support bounds that cross the antimeridian.'

export function toLatLngBounds (bounds: Bounds) {
  return new LatLngBounds([
    [bounds.north, bounds.west],
    [bounds.south, bounds.east]
  ])
}

/**
 * NB: We do not currently support crossing the antimeridian.
 */
export function fromLatLngBounds (bounds: LatLngBounds) {
  // Don't change the original bounds object
  let b = new LatLngBounds()
  b.extend(bounds)

  if (b.getWest() < -180) {
    b = new LatLngBounds(
      new LatLng(b.getSouth(), -180),
      b.getNorthEast()
    )

    window.alert(message('antimeridian', ANTIMERIDIAN_WARNING))
  }

  if (b.getEast() > 180) {
    b = new LatLngBounds(
      new LatLng(b.getSouth(), 180),
      b.getNorthEast()
    )
    window.alert(message('antimeridian', ANTIMERIDIAN_WARNING))
  }

  const bboxString = b.toBBoxString()
  return {
    north: b.getNorth(),
    south: b.getSouth(),
    east: b.getEast(),
    west: b.getWest(),
    toString: () => bboxString
  }
}

/**
 * Reproject bounds to the grid size that we use.
 */
export function reprojectBounds (bounds: LatLngBounds) {
  return new LatLngBounds(
    reprojectCoordinates(bounds.getSouthWest()),
    reprojectCoordinates(bounds.getNorthEast())
  )
}
