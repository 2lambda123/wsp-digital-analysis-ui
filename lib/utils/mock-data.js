import multi from '@conveyal/woonerf/store/multi'
import promise from '@conveyal/woonerf/store/promise'
import {middleware as fetchMiddleware} from '@conveyal/woonerf/fetch'
import configureStore from 'redux-mock-store'
import thunkMiddleware from 'redux-thunk'

import {ADD_TRIP_PATTERN} from '../constants'

export const makeMockStore = configureStore([fetchMiddleware, multi, promise, thunkMiddleware])

export const mockSegmentGeom = {
  type: 'LineString',
  coordinates: [
    [
      -122.0246,
      36.9707
    ],
    [
      -122.0279,
      37.049
    ],
    [
      -121.9799,
      37.2299
    ],
    [
      -121.9445,
      37.324
    ],
    [
      -121.936,
      37.353
    ],
    [
      -121.924,
      37.365
    ]
  ]
}

export const mockSegment = {
  type: 'Feature',
  properties: {},
  geometry: mockSegmentGeom
}

const mockStop1 = {
  stop_id: '1',
  stop_lat: 36.9707,
  stop_lon: -122.0246,
  stop_name: 'Santa Cruz'
}

const mockStop2 = {
  stop_id: '2',
  stop_lat: 37.049,
  stop_lon: -122.0279,
  stop_name: 'Scotts Valley'
}

export const mockStops = [mockStop1, mockStop2]

export const mockPattern = {
  geometry: mockSegmentGeom,
  stops: mockStops,
  trips: [
    {
      trip_id: 'abcd',
      start_time: 12345,
      trip_short_name: 'The Express',
      trip_headsign: 'To Downtown',
      duration: 1234
    }
  ]
}

export const mockRoute = {
  id: 'route1',
  patterns: [mockPattern]
}

export const mockFeed = {
  id: '1',
  routes: [mockRoute],
  routesById: {
    route1: mockRoute
  },
  stopsById: {
    '1': mockStop1,
    '2': mockStop2
  }
}

export const mockMapState = {
  allowExtend: true,
  extendFromEnd: true,
  followRoad: true,
  state: 'add-trip-pattern',
  modificationId: '1234'
}

export const mockTimetable = {
  name: 'Test timetable',
  segmentSpeeds: [],
  dwellTime: 10,
  monday: true,
  tuesday: true,
  wednesday: true,
  thursday: true,
  friday: true,
  saturday: false,
  sunday: false,
  headwaySecs: 900, // 15 minutes
  startTime: 28800, // 8am
  endTime: 57600, // 4pm
  patternTrips: ['abcd'],
  sourceTrip: 'abcd'
}

export const mockModification = {
  bidirectional: false,
  entries: [],
  feed: mockFeed.id,
  id: '1234',
  name: 'Test Modification',
  routes: ['route1'],
  segments: [],
  showOnMap: false,
  timetables: [mockTimetable],
  trips: ['abcd'],
  type: ADD_TRIP_PATTERN,
  variants: [0]
}

export const mockProject = {
  bounds: {
    north: 39,
    east: -76,
    south: 38,
    west: -77
  },
  description: 'Project description',
  id: '1',
  name: 'Mock Project'
}

export const mockScenario = {
  bundleId: '1',
  feeds: [mockFeed],
  feedsById: {
    [mockFeed.id]: mockFeed
  },
  id: '1',
  name: 'Mock Scenario',
  projectId: '1',
  variants: ['Default']
}

export const mockStores = {
  init: {
    analysis: {
      isFetchingIsochrone: false,
      isochroneCutoff: 60
    },
    mapState: {
      activeTrips: [],
      center: {
        lat: 38.8886,
        lon: -77.043
      },
      components: [],
      zoom: 12
    },
    network: {
      error: null,
      outstandingRequests: 0
    },
    project: {
      currentProject: null,
      projects: [],
      projectsById: {}
    },
    routing: {
      locationBeforeTransitions: null
    },
    scenario: {
      bundles: [],
      bundlesById: {},
      currentBundle: {
        id: '1'
      },
      currentScenario: mockScenario,
      feeds: [mockFeed],
      feedsById: {
        [mockFeed.id]: mockFeed
      },
      modifications: [mockModification],
      modificationsById: {
        [mockModification.id]: mockModification
      },
      modificationsByType: {
        [mockModification.type]: [mockModification]
      },
      scenarios: [],
      scenariosById: {},
      variants: ['Default']
    },
    user: {
      idToken: null
    }
  }
}

export const mockScenarioApplicationError = {
  'modificationId': 'Add Trip Pattern',
  'title': "Error while applying the modification entitled 'Add Trip Pattern'.",
  'messages': [
    'Headway is not greater than zero.'
  ]
}

export const mockProfileRequest = {
  date: '2016-01-16',
  fromTime: 25200,
  toTime: 32400,
  accessModes: 'WALK',
  directModes: 'WALK',
  egressModes: 'WALK',
  transitModes: 'TRANSIT',
  walkSpeed: 1.3888888888888888,
  bikeSpeed: 4.166666666666667,
  carSpeed: 20,
  streetTime: 90,
  maxWalkTime: 20,
  maxBikeTime: 20,
  maxCarTime: 45,
  minBikeTime: 10,
  minCarTime: 10,
  suboptimalMinutes: 5,
  reachabilityThreshold: 0,
  bikeSafe: 1,
  bikeSlope: 1,
  bikeTime: 1,
  bikeTrafficStress: 4,
  monteCarloDraws: 200,
  maxRides: 4
}

export const mockRegionalAnalyses = [
  {
    zoom: 9,
    width: 300,
    height: 300,
    north: 20024,
    west: 20015,
    name: 'ANALYSIS 1',
    grid: 'Jobs_total',
    scenarioId: '1234',
    variant: 0,
    bundleId: '5678',
    cutoffMinutes: 60,
    id: 'abcd',
    request: mockProfileRequest
  },
  {
    zoom: 9,
    width: 300,
    height: 300,
    north: 20024,
    west: 20015,
    name: 'ANALYSIS 2',
    grid: 'Jobs_total',
    scenarioId: '4321',
    variant: 1,
    bundleId: '5678',
    cutoffMinutes: 60,
    id: 'efgh',
    request: mockProfileRequest
  }
]

export const mockGrid = {
  zoom: 9,
  width: 300,
  height: 300,
  north: 20024,
  west: 20015,
  data: [],
  min: 0,
  max: 2000
}

export const mockBookmark = {
  name: 'Bookmark',
  profileRequest: mockProfileRequest,
  isochroneCutoff: 42,
  destinationGrid: 'Workers_total',
  projectId: 'PROJECT'
}
