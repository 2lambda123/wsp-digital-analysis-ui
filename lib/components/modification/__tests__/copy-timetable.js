import enzyme from 'enzyme'
import clone from 'lodash/cloneDeep'
import React from 'react'

import {mockModification, mockSegment} from 'lib/utils/mock-data'
import CopyTimetable from '../copy-timetable'

const mockRegionId = 'r1'
const mockRegionIdWithNoTimetables = 'r2'

const mockProject = {
  _id: 'p1',
  name: 'mock project'
}

// response only includes regions, projects and modifications that
// all have at least one timetable
const mockResponse = [
  {
    projects: [
      {
        ...mockProject,
        modifications: [
          {
            timetables: [
              {
                _id: 't1',
                monday: true,
                tuesday: true,
                wednesday: true,
                thursday: true,
                friday: true,
                saturday: false,
                sunday: false,
                name: 'M-F',
                startTime: 25200,
                endTime: 79200,
                headwaySecs: 600,
                exactTimes: false,
                phaseAtStop: null,
                phaseFromTimetable: null,
                phaseFromStop: null,
                phaseSeconds: 0,
                dwellTime: 0,
                segmentSpeeds: [16],
                dwellTimes: []
              }
            ],
            name: 'mock modification',
            _id: 'm1',
            segments: [mockSegment]
          }
        ]
      }
    ],
    name: 'mock region',
    _id: mockRegionId
  }
]

function timeoutPromise() {
  return new Promise(resolve => {
    setTimeout(resolve, 5)
  })
}

describe('Component > Modification > CopyTimetable', () => {
  it('renders correctly', async () => {
    const create = jest.fn()
    const wrapper = enzyme.shallow(
      <CopyTimetable
        create={create}
        currentModification={mockModification}
        currentProject={mockProject}
        regionId={mockRegionId}
        getTimetables={() => Promise.resolve(mockResponse)}
      />
    )

    // Look for an element that should only exist once loaded
    expect(wrapper.find({label: 'Timetable'}).exists()).toBeFalsy()

    // wait for load to finish
    await timeoutPromise()

    // Look for an existsing element
    expect(wrapper.find({label: 'Timetable'}).exists()).toBeTruthy()

    // Snapshot the loaded state
    expect(wrapper).toMatchSnapshot()

    // Click the button
    wrapper.find('Button').simulate('click')

    // Verify `create` was called
    expect(create).toHaveBeenCalled()

    // Unmount wihtout errors
    wrapper.unmount()
  })

  // if rendering a region without any timetables, show the first
  // timetable that is available
  it('renders with a region without timetables', async () => {
    const wrapper = enzyme.shallow(
      <CopyTimetable
        create={jest.fn()}
        currentModification={mockModification}
        currentProject={mockProject}
        regionId={mockRegionIdWithNoTimetables}
        getTimetables={() => Promise.resolve(mockResponse)}
      />
    )

    // wait for load to finish
    await timeoutPromise()

    // snapshot loaded state
    expect(wrapper).toMatchSnapshot()

    wrapper.unmount()
  })

  // a test case for when no timetables are available to copy
  it('renders when no timetables exist in database', async () => {
    const wrapper = enzyme.shallow(
      <CopyTimetable
        create={jest.fn()}
        currentModification={mockModification}
        currentProject={mockProject}
        regionId={mockRegionIdWithNoTimetables}
        getTimetables={() => Promise.resolve([])}
      />
    )

    // Wait for load to finish
    await timeoutPromise()

    // snapshot loaded state
    expect(wrapper).toMatchSnapshot()

    // unmount
    wrapper.unmount()
  })

  describe('segment warnings', () => {
    it('shows a warning when current modification has no segments', async () => {
      const wrapper = enzyme.shallow(
        <CopyTimetable
          create={jest.fn()}
          currentModification={{
            ...mockModification,
            segments: []
          }}
          currentProject={mockProject}
          regionId={mockRegionId}
          getTimetables={() => Promise.resolve(mockResponse)}
        />
      )

      // wait for load to finish
      await timeoutPromise()

      // snapshot loaded state
      expect(wrapper).toMatchSnapshot()

      wrapper.unmount()
    })

    it('shows a warning when modification of timetable being copied has no segments', async () => {
      const serverResponse = clone(mockResponse)
      serverResponse[0].projects[0].modifications[0].segments = []
      serverResponse[0].projects[0].modifications[0].timetables[0].segmentSpeeds = []

      const wrapper = enzyme.shallow(
        <CopyTimetable
          create={jest.fn()}
          currentModification={mockModification}
          currentProject={mockProject}
          regionId={mockRegionId}
          getTimetables={() => Promise.resolve(serverResponse)}
        />
      )

      // wait for load to finish
      await timeoutPromise()

      // snapshot loaded state
      expect(wrapper).toMatchSnapshot()

      wrapper.unmount()
    })

    it('shows a warning when modification of timetable being copied has more segments', async () => {
      const serverResponse = clone(mockResponse)
      serverResponse[0].projects[0].modifications[0].segments.push(mockSegment)
      serverResponse[0].projects[0].modifications[0].timetables[0].segmentSpeeds.push(
        17
      )

      const wrapper = enzyme.shallow(
        <CopyTimetable
          create={jest.fn()}
          currentModification={mockModification}
          currentProject={mockProject}
          regionId={mockRegionId}
          getTimetables={() => Promise.resolve(serverResponse)}
        />
      )

      // wait for load to finish
      await timeoutPromise()

      // snapshot loaded state
      expect(wrapper).toMatchSnapshot()

      wrapper.unmount()
    })

    it('shows a warning when modification of timetable being copied has less segments', async () => {
      const segments = [...mockModification.segments, mockSegment]

      const wrapper = enzyme.shallow(
        <CopyTimetable
          create={jest.fn()}
          currentModification={{
            ...mockModification,
            segments
          }}
          currentProject={mockProject}
          regionId={mockRegionId}
          getTimetables={() => Promise.resolve(mockResponse)}
        />
      )

      // wait for load to finish
      await timeoutPromise()

      // snapshot loaded state
      expect(wrapper).toMatchSnapshot()

      wrapper.unmount()
    })
  })
})
