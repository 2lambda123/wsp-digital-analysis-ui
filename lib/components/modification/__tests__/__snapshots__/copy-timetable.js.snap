// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > Modification > CopyTimetable renders correctly 1`] = `
<ModalBody>
  <Select
    label="Region"
    onBlur={[Function]}
    onChange={[Function]}
    value="r1"
  >
    <option
      key="regionOption-r1"
      value="r1"
    >
      mock region
    </option>
  </Select>
  <Select
    label="Project"
    onBlur={[Function]}
    onChange={[Function]}
    value="p1"
  >
    <option
      key="projectOption-p1"
      value="p1"
    >
      mock project
    </option>
  </Select>
  <Select
    label="Modification"
    onBlur={[Function]}
    onChange={[Function]}
    value="m1"
  >
    <option
      key="modificationOption-m1"
      value="m1"
    >
      mock modification
    </option>
  </Select>
  <Select
    label="Timetable"
    onBlur={[Function]}
    onChange={[Function]}
    value="t1"
  >
    <option
      key="timetableOption-t1"
      value="t1"
    >
      M-F
    </option>
  </Select>
  <P>
    <Button
      block={true}
      onClick={[Function]}
      style="success"
    >
      <Icon
        icon={
          Object {
            "icon": Array [],
            "iconName": "plus",
            "prefix": "fas",
          }
        }
      />
       Copy into new timetable
    </Button>
  </P>
</ModalBody>
`;

exports[`Component > Modification > CopyTimetable renders when no timetables exist in database 1`] = `
<ModalBody>
  <div
    className="alert alert-danger"
    role="alert"
  >
    <Icon
      icon={
        Object {
          "icon": Array [],
          "iconName": "exclamation-circle",
          "prefix": "fas",
        }
      }
    />
    No timetables available to copy from! Please create a timetable manually.
  </div>
</ModalBody>
`;

exports[`Component > Modification > CopyTimetable renders with a region without timetables 1`] = `
<ModalBody>
  <Select
    label="Region"
    onBlur={[Function]}
    onChange={[Function]}
    value="r1"
  >
    <option
      key="regionOption-r1"
      value="r1"
    >
      mock region
    </option>
  </Select>
  <Select
    label="Project"
    onBlur={[Function]}
    onChange={[Function]}
    value="p1"
  >
    <option
      key="projectOption-p1"
      value="p1"
    >
      mock project
    </option>
  </Select>
  <Select
    label="Modification"
    onBlur={[Function]}
    onChange={[Function]}
    value="m1"
  >
    <option
      key="modificationOption-m1"
      value="m1"
    >
      mock modification
    </option>
  </Select>
  <Select
    label="Timetable"
    onBlur={[Function]}
    onChange={[Function]}
    value="t1"
  >
    <option
      key="timetableOption-t1"
      value="t1"
    >
      M-F
    </option>
  </Select>
  <P>
    <Button
      block={true}
      onClick={[Function]}
      style="success"
    >
      <Icon
        icon={
          Object {
            "icon": Array [],
            "iconName": "plus",
            "prefix": "fas",
          }
        }
      />
       Copy into new timetable
    </Button>
  </P>
</ModalBody>
`;

exports[`Component > Modification > CopyTimetable segment warnings shows a warning when current modification has no segments 1`] = `
<ModalBody>
  <Select
    label="Region"
    onBlur={[Function]}
    onChange={[Function]}
    value="r1"
  >
    <option
      key="regionOption-r1"
      value="r1"
    >
      mock region
    </option>
  </Select>
  <Select
    label="Project"
    onBlur={[Function]}
    onChange={[Function]}
    value="p1"
  >
    <option
      key="projectOption-p1"
      value="p1"
    >
      mock project
    </option>
  </Select>
  <Select
    label="Modification"
    onBlur={[Function]}
    onChange={[Function]}
    value="m1"
  >
    <option
      key="modificationOption-m1"
      value="m1"
    >
      mock modification
    </option>
  </Select>
  <Select
    label="Timetable"
    onBlur={[Function]}
    onChange={[Function]}
    value="t1"
  >
    <option
      key="timetableOption-t1"
      value="t1"
    >
      M-F
    </option>
  </Select>
  <div
    className="alert alert-info"
    role="alert"
  >
    <Icon
      icon={
        Object {
          "icon": Array [],
          "iconName": "exclamation-circle",
          "prefix": "fas",
        }
      }
    />
    The modification being copied to and/or from does not have any segments. The default segment speed of 15 km/h will be used.
  </div>
  <P>
    <Button
      block={true}
      onClick={[Function]}
      style="success"
    >
      <Icon
        icon={
          Object {
            "icon": Array [],
            "iconName": "plus",
            "prefix": "fas",
          }
        }
      />
       Copy into new timetable
    </Button>
  </P>
</ModalBody>
`;

exports[`Component > Modification > CopyTimetable segment warnings shows a warning when modification of timetable being copied has less segments 1`] = `
<ModalBody>
  <Select
    label="Region"
    onBlur={[Function]}
    onChange={[Function]}
    value="r1"
  >
    <option
      key="regionOption-r1"
      value="r1"
    >
      mock region
    </option>
  </Select>
  <Select
    label="Project"
    onBlur={[Function]}
    onChange={[Function]}
    value="p1"
  >
    <option
      key="projectOption-p1"
      value="p1"
    >
      mock project
    </option>
  </Select>
  <Select
    label="Modification"
    onBlur={[Function]}
    onChange={[Function]}
    value="m1"
  >
    <option
      key="modificationOption-m1"
      value="m1"
    >
      mock modification
    </option>
  </Select>
  <Select
    label="Timetable"
    onBlur={[Function]}
    onChange={[Function]}
    value="t1"
  >
    <option
      key="timetableOption-t1"
      value="t1"
    >
      M-F
    </option>
  </Select>
  <div
    className="alert alert-info"
    role="alert"
  >
    <Icon
      icon={
        Object {
          "icon": Array [],
          "iconName": "exclamation-circle",
          "prefix": "fas",
        }
      }
    />
    The modification of the timetable you are copying from has less segments than the modification you are copying to. The segment speeds will be copied 1 segment(s) and after that, the default segment speed of 15 km/h will be used.
  </div>
  <P>
    <Button
      block={true}
      onClick={[Function]}
      style="success"
    >
      <Icon
        icon={
          Object {
            "icon": Array [],
            "iconName": "plus",
            "prefix": "fas",
          }
        }
      />
       Copy into new timetable
    </Button>
  </P>
</ModalBody>
`;

exports[`Component > Modification > CopyTimetable segment warnings shows a warning when modification of timetable being copied has more segments 1`] = `
<ModalBody>
  <Select
    label="Region"
    onBlur={[Function]}
    onChange={[Function]}
    value="r1"
  >
    <option
      key="regionOption-r1"
      value="r1"
    >
      mock region
    </option>
  </Select>
  <Select
    label="Project"
    onBlur={[Function]}
    onChange={[Function]}
    value="p1"
  >
    <option
      key="projectOption-p1"
      value="p1"
    >
      mock project
    </option>
  </Select>
  <Select
    label="Modification"
    onBlur={[Function]}
    onChange={[Function]}
    value="m1"
  >
    <option
      key="modificationOption-m1"
      value="m1"
    >
      mock modification
    </option>
  </Select>
  <Select
    label="Timetable"
    onBlur={[Function]}
    onChange={[Function]}
    value="t1"
  >
    <option
      key="timetableOption-t1"
      value="t1"
    >
      M-F
    </option>
  </Select>
  <div
    className="alert alert-info"
    role="alert"
  >
    <Icon
      icon={
        Object {
          "icon": Array [],
          "iconName": "exclamation-circle",
          "prefix": "fas",
        }
      }
    />
    The modification of the timetable you are copying to has less segments than the modification you are copying from. The segment speeds will be copied 1 segment(s).
  </div>
  <P>
    <Button
      block={true}
      onClick={[Function]}
      style="success"
    >
      <Icon
        icon={
          Object {
            "icon": Array [],
            "iconName": "plus",
            "prefix": "fas",
          }
        }
      />
       Copy into new timetable
    </Button>
  </P>
</ModalBody>
`;

exports[`Component > Modification > CopyTimetable segment warnings shows a warning when modification of timetable being copied has no segments 1`] = `
<ModalBody>
  <Select
    label="Region"
    onBlur={[Function]}
    onChange={[Function]}
    value="r1"
  >
    <option
      key="regionOption-r1"
      value="r1"
    >
      mock region
    </option>
  </Select>
  <Select
    label="Project"
    onBlur={[Function]}
    onChange={[Function]}
    value="p1"
  >
    <option
      key="projectOption-p1"
      value="p1"
    >
      mock project
    </option>
  </Select>
  <Select
    label="Modification"
    onBlur={[Function]}
    onChange={[Function]}
    value="m1"
  >
    <option
      key="modificationOption-m1"
      value="m1"
    >
      mock modification
    </option>
  </Select>
  <Select
    label="Timetable"
    onBlur={[Function]}
    onChange={[Function]}
    value="t1"
  >
    <option
      key="timetableOption-t1"
      value="t1"
    >
      M-F
    </option>
  </Select>
  <div
    className="alert alert-info"
    role="alert"
  >
    <Icon
      icon={
        Object {
          "icon": Array [],
          "iconName": "exclamation-circle",
          "prefix": "fas",
        }
      }
    />
    The modification being copied to and/or from does not have any segments. The default segment speed of 15 km/h will be used.
  </div>
  <P>
    <Button
      block={true}
      onClick={[Function]}
      style="success"
    >
      <Icon
        icon={
          Object {
            "icon": Array [],
            "iconName": "plus",
            "prefix": "fas",
          }
        }
      />
       Copy into new timetable
    </Button>
  </P>
</ModalBody>
`;
